packages/network-controller/src/NetworkController.ts

```diff
@@ -1354,19 +1352,20 @@ export class NetworkController extends BaseController<
    * removed in a future release
    */
   async setProviderType(type: InfuraNetworkType) {
-    assert.notStrictEqual(
-      type,
-      NetworkType.rpc,
-      // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-      `NetworkController - cannot call "setProviderType" with type "${NetworkType.rpc}". Use "setActiveNetwork"`,
-    );
-    assert.ok(
-      isInfuraNetworkType(type),
-      // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-      `Unknown Infura provider type "${type}".`,
-    );
+    if ((type as unknown) === NetworkType.rpc) {
+      throw new Error(
+        // TODO: Either fix this lint violation or explain why it's necessary to ignore.
+        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+        `NetworkController - cannot call "setProviderType" with type "${NetworkType.rpc}". Use "setActiveNetwork"`,
+      );
+    }
+    if (!isInfuraNetworkType(type)) {
+      throw new Error(
+        // TODO: Either fix this lint violation or explain why it's necessary to ignore.
+        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+        `Unknown Infura provider type "${type}".`,
```

Comment:
Nit: We should probably wrap `type` in `String(...)` here to fix this error. This one seems legitimate.

TypeScript is complaining because we've narrowed `type` here to `never`, because if the value matches the type, this should be unreachable. It's only reachable if the wrong type is passed in. Given that, maybe we shouldn't assume it's a string, it could be anything.

Pre-existing issue though.

Reply:
I'm updating these lines, so I might as well fix this too. Done in [557c1a4](https://github.com/MetaMask/core/pull/3672/commits/557c1a4a31e9e33e9fe8f2a7c063901fb1798856).